tutorials/introductory/patchwork_axes.py

```diff
@@ -0,0 +1,195 @@
+Xb
+""",
+        empty_sentinel="X",
+    )
+)
+
+
+###############################################################################
+# This can be particularly useful if your editor strips trailing white space.
+#
+# Because these are there are string inputs any Unicode glyph is valid!
+
+label_axes(
+    plt.figure(constrained_layout=True).build_grid(
+        """
+αб
+ℝ☢
+"""
+    )
+)
+
+###############################################################################
+# This feature is built on top of `.gridspec` and you can pass the
+# same keyword arguments through (the same with `.Figure.subplots` as
+# well) see `.gridspec.GridSpec` for the full set of keywords
+
+label_axes(
+    plt.figure(constrained_layout=True).build_grid(
+        """
+xax
+bAc
+xdx
+""",
+        gridspec_kw={"height_ratios": [1, 3, 1], "width_ratios": [1, 3, 1]},
+        empty_sentinel="x",
+    )
+)
+
+###############################################################################
+# You can also pass through arguments used to create the subplots
+# (again, the same as `.Figure.subplots`).
+
+label_axes(
+    plt.figure(constrained_layout=True).build_grid(
+        "AB", subplot_kw={"projection": "polar"}
+    )
+)
+
+
+###############################################################################
+# List input
+# ==========
+#
+# While the mulit-line string specification covers many use cases, we may want
+# to use more than a single letter to label our axes.
+
+axd = label_axes(
+    plt.figure(constrained_layout=True).build_grid(
+        [["main", "main", "callout"], ["main", "main", "BLANK"]],
+        empty_sentinel="BLANK",
+    )
+)
+
+
+###############################################################################
+# which lets us give the Axes meaningful labels.  Using this you can access the
+# axes by name ::
+#
+#   make_plot(axd['main'])
+#   axd['callout'].plot(x, y)
+#
+# which can make your code more readable and resilient to changing the layout.
+#
+# Using the list specification you can pass in any hashable object (things
+# that can be the key in a `dict`).  In the special case of the objects being
+# existing `.axes.Axes` object we will re-arrange them to fit.
+
+fig, (ax1, ax2, ax3) = plt.subplots(3, tight_layout=True)
+fig.build_grid([[ax1, ax1], [ax2, ax3]])
+ax1.set_title("top", loc="left")
+ax2.set_title("bottom right", loc="left")
+ax3.set_title("bottom right", loc="left")
+
+
+###############################################################################
+# You can also pass in a 2D numpy array to do things like
+np_layout = np.zeros((4, 4), dtype=int)
+for j in range(4):
+    np_layout[j, j] = j + 1
+label_axes(
+    plt.figure(constrained_layout=True).build_grid(np_layout, empty_sentinel=0)
+)
+
+
+###############################################################################
+# Using the list input we can also nest the specification
+
+inner = [["inner top"], ["inner bottom"]]
+
+outer = [["main", inner], ["bottom", "bottom"]]
```

Comment:
Again, break these to make the parallel to the figure clearer.
```suggestion
inner = [
    ["inner top"],
    ["inner bottom"],
]

outer = [
    ["main", inner],
    ["bottom", "bottom"],
]
```